data-prepper-api/src/test/java/com/amazon/dataprepper/model/event/JacksonEventTest.java

```diff
@@ -463,4 +481,10 @@ public void testBuild_withInvalidStringData() {
         assertThrows(IllegalArgumentException.class, () -> builder.build());
     }
 
+    @Test
+    public void testTimestamp_isNotNullAndValid() {
+        Assertions.assertNotNull(timestamp);
+        Assertions.assertTrue(Duration.between(timestamp, OffsetDateTime.now()).getSeconds() < 1);
```

Comment:
Is there ever a chance that all the tests take more than one second to run before this assertion is made? If so this could result in a flaky test. It may make more sense to make the timestamp here right before the assertion

Reply:
Good catch. But, the `timestamp` is not created separately, we are getting it from the event which will be created for each test. So technically it shouldn't take more than 1 second.

Reply:
This is another reason why I like `createObjectUnderTest()` methods instead of creating it in `setUp()`.

```
Instant before = Instant.now();
Event objectUnderTest = createObjectUnderTest();
Instant after = Instant.now();

assertThat(objectUnderTest.get("event_timestamp"), lessThanOrEqualTo(after));
assertThat(objectUnderTest.get("event_timestamp"), greaterThanOrEqualTo(before));
```